File: LATIHAN1DPBO2025C2/CPP/program/Main.cpp
/* Saya Khana Yusdiana 2100991 mengerjakan soal Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan
seperti yang telah dispesifikasikan. Aamiin. */

// import library dan file
#include <bits/stdc++.h>
#include "Crud.cpp"

// menggunakan standarad namespace
using namespace std;

int main()
{
    // deklarasi variabel
    Crud crud;                                             // Intansiasi objek crud
    string id, nama_produk, kategori_produk, harga_produk; // variabel untuk menampung string inputan
    int m;                                                 // variabel untuk memilih menu crud
    do
    {
        // menampilkan menu
        cout << "\nMenu : " << '\n';
        cout << "0. Exit" << '\n';
        cout << "1. Create" << '\n';
        cout << "2. Read" << '\n';
        cout << "3. Update" << '\n';
        cout << "4. Delete" << '\n';

        cout << "\nSilahkan pilih menu yang diinginkan : ";
        cin >> m; // input menu pilihan

        if (m == 1) // jika menu yang dipilih adalah 1
        {
            // meminta masukan data objek produk
            cout << "\nMasukan data produk : \n";
            cout << "id: ";
            cin >> id;
            cout << "Nama Produk: ";
            cin >> nama_produk;
            cout << "Kategori Produk: ";
            cin >> kategori_produk;
            cout << "Harga Produk: ";
            cin >> harga_produk;
            // memanggil method create yang ada pada objek crud
            crud.create(id, nama_produk, kategori_produk, harga_produk);
        }
        else if (m == 2) // jika menu yang dipilih adalah 2
        {
            // memanggil method read yang ada pada objek crud
            crud.read();
        }
        else if (m == 3) // jika menu yang dipilih adalah 3
        {
            // input id produk yang data nya akan di update
            cout << "\nMasukan id produk yang akan di update : ";
            cin >> id;
            // memanggil method update yang ada pada objek crud
            crud.update(id);
        }
        else if (m == 4) // jika menu yang dipilih adalah 4
        {
            // input id produk yang data nya akan di delete
            cout << "\nMasukan id produk yang akan di delete : ";
            cin >> id;
            // memanggil method delete yang ada pada objek crud
            crud.del(id);
        }
        else if (m == 0) // jika menu yang dipilih adalah 0
        {
            cout << "\nTerima Kasih" << '\n';
        }
    } while (m != 0); // perulangan berjalan selama user tidak menginput 0

    return 0;
}

File: LATIHAN1DPBO2025C2/Python/program/main.py
# Saya Khana Yusdiana NIM 2100991 mengerjakan soal Latihan 1
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan
# seperti yang telah dispesifikasikan. Aamiin. 

# import class file
from Crud import Crud

# Intansiasi objek crud
crud = Crud()

# deklarasi variabel untuk menampung string inputan
id = ""
nama_produk = ""
kategori_produk = ""
harga_produk = ""

# variabel untuk memilih menu crud
n = 5
while n != 0:
    # menampilkan menu
    print("\nMenu: ")
    print("0. Exit")
    print("1. Create")
    print("2. Read")
    print("3. Update")
    print("4. Delete")
    print("\nSilahkan pilih menu yang diinginkan: ")
    # input menu pilihan
    n = int(input())
    # jika menu yang dipilih adalah 1
    if (n == 1):
        # meminta masukan data objek mahasiswa
        print("\nMasukan data yang ingin dibuat !")
        print("Id: ")
        id = str(input())
        print("Nama Produk : ")
        nama_produk = str(input())
        print("Kategori Produk: ")
        kategori_produk = str(input())
        print("Harga Produk: ")
        harga_produk = str(input())
        # memanggil method create yang ada pada objek crud
        crud.create(id, nama_produk, kategori_produk, harga_produk)
    # jika menu yang dipilih adalah 2
    elif (n == 2):
        crud.read()  # memanggil method read yang ada pada objek crud
    # jika menu yang dipilih adalah 3
    elif (n == 3):
        # input nim mahasiswa yang data nya akan di update
        print("\nMasukan Id yang akan di update : ")
        id = str(input())
        # memanggil method update yang ada pada objek crud
        crud.update(id)
    # jika menu yang dipilih adalah 4
    elif (n == 4):
        # input nim mahasiswa yang data nya akan di update
        print("\nMasukan Id yang akan di delete : ")
        id = str(input())
        # memanggil method update yang ada pada objek crud
        crud.delete(id)
    # jika menu yang dipilih adalah 0
    elif (n == 0):
        print("\nTerima Kasih")
